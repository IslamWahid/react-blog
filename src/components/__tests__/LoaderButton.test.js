import React from 'react';
import LoaderButton from '../LoaderButton';
import { shallow } from 'enzyme';

describe('LoaderButton Component', () => {
  let loaderButton;

  beforeEach(() => {
    loaderButton = shallow(
      <LoaderButton
        block
        bsStyle="primary"
        bsSize="large"
        type="submit"
        isLoading={false}
        text="Create"
        loadingText="Creatingâ€¦"
      />
    );
  });
  it('should match snapshot', () => {
    expect(loaderButton).toMatchSnapshot();
  });
});
