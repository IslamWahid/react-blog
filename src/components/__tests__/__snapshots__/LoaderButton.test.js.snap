// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderButton Component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    block={true}
    bsSize="large"
    bsStyle="primary"
    isLoading={false}
    loadingText="Creatingâ€¦"
    text="Create"
    type="submit"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "active": false,
      "block": true,
      "bsClass": "btn",
      "bsSize": "large",
      "bsStyle": "primary",
      "children": Array [
        false,
        "Create",
      ],
      "className": "LoaderButton ",
      "disabled": false,
      "type": "submit",
    },
    "ref": null,
    "rendered": Array [
      false,
      "Create",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": false,
        "block": true,
        "bsClass": "btn",
        "bsSize": "large",
        "bsStyle": "primary",
        "children": Array [
          false,
          "Create",
        ],
        "className": "LoaderButton ",
        "disabled": false,
        "type": "submit",
      },
      "ref": null,
      "rendered": Array [
        false,
        "Create",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
